cypress/e2e/GUI/ui-funcionality-tests.cy.js

```diff
@@ -0,0 +1,155 @@
+describe('EngageSphere', () => {
+  beforeEach(() => {
+    Cypress.on('window:before:load', window => {
+      window.document.cookie = 'cookieConsent=accepted'
+    })
+    cy.visit('/');
+  });
+
+  it('Should maintain filters when returning from customer details', () => {
+    cy.get('[data-testid="size-filter"]')
+      .select('Small');
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+      cy.contains('button', 'Back')
+      .click();
+    cy.get('[data-testid="size-filter"]')
+      .should('have.value', 'Small');
+  });
+
+  it('Should return to the customer list after clicking back', () => {
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+    cy.contains('button', 'Back')
+      .click();
+    cy.get('[data-testid="table"]')
+      .should('be.visible');
+  });
+
+  it('Should display the footer with the correct text and links', () => {
+    cy.contains('p', 'Copyright 2024 - Talking About Testing')
+    .should('be.visible');
+  cy.contains('a', 'Hotmart')
+    .should('be.visible')
+    .and('have.attr', 'href', 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem');
+  cy.contains('a', 'Udemy')
+    .should('be.visible')
+    .and('have.attr', 'href', 'https://udemy.com/user/walmyr');
+  cy.contains('a', 'Blog')
+    .should('be.visible')
+    .and('have.attr', 'href', 'https://talkingabouttesting.com');
+  cy.contains('a', 'YouTube')
+    .should('be.visible')
+    .and('have.attr', 'href', 'https://youtube.com/@talkingabouttesting');
+  });
+
+  it('Should displays the greeting "Hi, there" when no name is given', () => {
+    cy.contains('h2', 'Hi there!')
+      .should('be.visible');
+  });
+
+  it('Should displays the greeting "Hi, Joe" when the name Joe is given', () => {
+    cy.get('input[type="text"]')
+      .type('Joe');
```

Comment:
```suggestion
      .type('Joe');

```

Só separando ação de resultado esperado.

Reply:
Aplique o mesmo aos outros testes.

E antes de fazer tal separação, pense bem sobre quais são as pré-condições dos testes, suas ações e seus resultados esperados.